src/loaders/svg/tvgSvgSceneBuilder.cpp

```diff
@@ -396,13 +396,9 @@ static bool _recognizeShape(SvgNode* node, Shape* shape)
     switch (node->type) {
         case SvgNodeType::Path: {
             if (node->node.path.path) {
-                Array<PathCommand> cmds;
-                Array<Point> pts;
-                if (svgPathToTvgPath(node->node.path.path, cmds, pts)) {
-                    if (shape->appendPath(cmds.data, cmds.count, pts.data, pts.count) != Result::Success) {
-                        TVGERR("SVG", "Invalid path information.");
-                        return false;
-                    }
+                if (!svgPathToShape(node->node.path.path, shape)) {
```

Comment:
tvgSvgPath only accessed data related to Path. This way causes unnecessary includes.
I think below is better than modifying the function parameters. what do you think?

```suggestion
auto& pts = P(shape)->rs.path.pts;
auto& cmds = P(shape)->rs.path.cmds;
   if (!svgPathToTvgPath(node->node.path.path, cmds, pts)) {
...

Reply:
Could you please suggest the beneficial point? Either way it will be a thorvg specific internal function. For now, less parameter, simpler function prototypes are better I think. or probably we can remove that function since nowhere uses it.

Reply:
Also, at least now, the helper function guides you to pass the Shape so that it will optimally append the data in the under the hood. Otherwise someone might call it again with the previous way with an unnecessary memory copy.

Reply:
> Could you please suggest the beneficial point?

1. No unnecessary header inclusion.(`#include "tvgShape.h"`) And appendPath() API is not used, memory copy does not occur.
2. `tvgSvgPath.cpp` code is basically responsible for translating data related to SVG Path. So we need is a Path String and the PathComand we passed it to. (I think this is why we separated this function, which was only used here, into a separate file.)
3. I think that having access to Shape objects is beyond the scope of tvgSvgPath's role.
 
> Otherwise someone might call it again with the previous way - an unnecessary memory copy.

Even if memory copy occurs when someone uses this API, it is because they use appendPath and not because of this API.

Anyway, I am not against to applying this patch because it does not have any functional problems.
I tested several expected paths and they work without problem.

Reply:
Ok, if you find out any real problems then please revise the code yourself. Otherwise let's keep it simple and easy code.